data/helpers/ingredientsDb.js

```diff
@@ -0,0 +1,21 @@
+const db = require('../../dbConfig')
+
+module.exports = {
+  getIngredients,
+  getIngredient,
+  addIngredient
+}
+
+function getIngredients() {
+  return db('ingredients')
+}
+function getIngredient(id) {
+  return db('ingredients')
+    .where({ id })
+    .first()
+}
+function addIngredient(body) {
+  return db('ingredients')
+    .insert(body)
+    .then(ids => ids[0])
```

Comment:
These helpers are identical to the ones for dishes, and also identical to 2/3 of the ones for recipes. Why not extract these out into some general helpers, and rename them in their separate files after the fact? For example, in a new file called something like generalHelper.js in your helpers directory:
```javascript
const db = require('../../dbConfig')

function makeHelpers(tableName) {
  function getAll() {
    return db(tableName)
  }

  function getOne(id) {
    return db(tableName)
      .where({ id })
      .first()
  }

  function addOne(body) {
    return db(tableName)
      .insert(body)
      .then(ids => ids[0])
  }

  return {
    getAll, getOne, addOne
  };
};

module.exports = makeHelpers;
```

Reply:
You could then use that in your specific files, and replace them as needed; here's ingredients as an example:
```javascript
const makeHelpers = require('./generalHelper.js');

const { getAll, getOne, addOne } = makeHelpers('ingredients');

module.exports = {
  getIngredients: getAll,
  getIngredient: getOne,
  addIngredient: addOne
};
```
For the [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring) on the third line, you could just as easily name the whole object `helpers` or something of the sort and just access each method off of that.

For recipes, you'd just ignore the `getOne` to replace it with the custom one for recipes. 😃